daemon/daemon.go

```diff
@@ -922,12 +922,29 @@ func (daemon *Daemon) GetRemappedUIDGID() (int, int) {
 	return uid, gid
 }
 
-// tempDir returns the default directory to use for temporary files.
-func tempDir(rootDir string, rootUID, rootGID int) (string, error) {
+// prepareTempDir prepares and returns the default directory to use
+// for temporary files.
+// If it doesn't exist, it is created. If it exists, its content is removed.
+func prepareTempDir(rootDir string, rootUID, rootGID int) (string, error) {
 	var tmpDir string
 	if tmpDir = os.Getenv("DOCKER_TMPDIR"); tmpDir == "" {
 		tmpDir = filepath.Join(rootDir, "tmp")
+		newName := tmpDir + "-old"
```

Comment:
If a daemon crashes before completely removing `tmpDir + "-old"`, the next daemon execution fails to execute `os.Rename(tmpDir, tmpDir+"-old")`, and hence the old `tmpDir + "-old"` will be left alone.

So I think we need `os.RemoveAll(tmpDir + "-old")` here.